src/snowflake/snowpark/_internal/type_utils.py

```diff
@@ -325,7 +325,7 @@ def infer_type(obj: Any) -> DataType:
 
 
 def infer_schema(
-    row: Union[Dict, List, Tuple], names: Optional[List] = None
+    row: Union[Dict, List, Tuple], names: Optional[Union[List, Tuple]] = None
```

Comment:
I don't think we could do `names.extend`. How about we explicitly copy `names` into a list first? This way we could support `Iterable`.

Reply:
You are right. I can convert iterable into a list and then assume list datatype from thereon...